### PR TITLE
usm: kafka: Add TLS support

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Add support for Kafka over TLS.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-147

#### `kafka.tls` Load test results

Base version is 7.54.0-rc.4 and the test type was `kafka.tls` with features set to `kafka,go-tls`.

system-probe max k8s CPU Usage  (2)
system-probe RSS (avg) (2)
system-probe working set (avg) (2)

### Describe how to test/QA your changes

Tests are included in the PR and have been added to load test.

### CODE REVIEW

**Comment by DanielLavie** [comment]
Could you provide a description of what specifically you're comparing in the load test, particularly in terms of agent configuration?

**Review by guyarb — CHANGES_REQUESTED** [review]
A general comment - that's a great PR. Very well divided into commits with good commit messages and it was very easy to dive into the code

File: pkg/network/ebpf/c/protocols/kafka/pktbuf.h [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    switch (pkt.type) {
+    case PKTBUF_SKB:
+        return bpf_skb_load_bytes(pkt.skb, offset, to, len);
+    case PKTBUF_TLS:
+        return bpf_probe_read_user(to, len, pkt.tls->buffer_ptr + offset);
+    }
+
+    pktbuf_invalid_operation();
+    return 0;
+}
```
[the remark this answers is not in the document]
Comment (vitkyrka) [reply pkg/network/ebpf/c/protocols/kafka/pktbuf.h]: Done.

**Comment by vitkyrka** [comment]
> Could you provide a description of what specifically you're comparing in the load test, particularly in terms of agent configuration?

I've added this information to the PR description now.

File: pkg/network/ebpf/c/protocols/tls/https.h [hunk 2]
```diff
@@ -107,6 +127,39 @@ static __always_inline void tls_process(struct pt_regs *ctx, conn_tuple_t *t, vo
...
+    }
+
+    protocol_stack_t *stack = get_protocol_stack(&normalized_tuple);
+    if (!stack) {
+        return;
+    }
+
+    set_protocol(stack, PROTOCOL_KAFKA);
+    bpf_tail_call_compat(ctx, &tls_process_progs, TLS_KAFKA);
+}
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/tls/https.h:161]: I know what's the problem here, I'll address it in a separate PR
I found the same during my work to modularize USM

[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/tls/https.h:161]: you can resolve the ticket

**Review by guyarb — APPROVED** [review]

**Pull request merged**